File: src/Prompts/Core/OutputFormats.php
<?php

declare(strict_types=1);

namespace App\Prompts\Core;

use App\PromptLoader;

/**
 * 出力フォーマット定義 - 分析結果の統一的な出力形式を定義
 */
final class OutputFormats
{
    private static ?PromptLoader $loader = null;

    private static function getLoader(): PromptLoader
    {
        if (self::$loader === null) {
            self::$loader = new PromptLoader();
        }
        return self::$loader;
    }

    /**
     * 関心の分離の欠陥テーブル形式
     */
    public static function separationOfConcernsTable(): string
    {
        return self::getLoader()->getContent('output-formats/separation-of-concerns-table');
    }

    /**
     * カプセル化の欠陥テーブル形式
     */
    public static function encapsulationTable(): string
    {
        return self::getLoader()->getContent('output-formats/encapsulation-table');
    }

    /**
     * ドメインモデル完全性の欠陥テーブル形式
     */
    public static function domainModelTable(): string
    {
        return self::getLoader()->getContent('output-formats/domain-model-table');
    }

    /**
     * レイヤ違反の欠陥テーブル形式
     */
    public static function layerViolationTable(): string
    {
        return self::getLoader()->getContent('output-formats/layer-violation-table');
    }

    /**
     * 設計改善提案テーブル形式
     */
    public static function refactoringSuggestionTable(): string
    {
        return self::getLoader()->getContent('output-formats/refactoring-suggestion-table');
    }

    /**
     * Mermaidクラス図の出力形式
     */
    public static function mermaidClassDiagram(): string
    {
        return self::getLoader()->getContent('output-formats/mermaid-class-diagram');
    }

    /**
     * スコアサマリー形式
     */
    public static function scoreSummary(): string
    {
        return self::getLoader()->getContent('output-formats/score-summary');
    }

    /**
     * 全ての出力フォーマットを結合
     */
    public static function all(): string
    {
        return implode("\n\n---\n\n", [
            self::separationOfConcernsTable(),
            self::encapsulationTable(),
            self::domainModelTable(),
            self::layerViolationTable(),
            self::scoreSummary(),
        ]);
    }

    /**
     * リファクタリング提案用の出力フォーマット
     */
    public static function forRefactoring(): string
    {
        return implode("\n\n---\n\n", [
            self::refactoringSuggestionTable(),
            self::mermaidClassDiagram(),
        ]);
    }
}

File: internal/domain/output_formats.go
package domain

import (
	"seiren/internal/promptloader"
	"strings"
)

// focusに対応する出力フォーマットのマッピング
var focusOutputFormats = map[Focus][]string{
	FocusEncapsulation:        {"output-formats/encapsulation-table"},
	FocusSeparationOfConcerns: {"output-formats/separation-of-concerns-table"},
	FocusDomainModel:          {"output-formats/domain-model-table"},
	FocusLayerSeparation:      {"output-formats/layer-violation-table"},
	FocusLayerDataModel:       {"output-formats/layer-violation-table"},
	FocusLayerRepository:      {"output-formats/layer-violation-table"},
	FocusLayerService:         {"output-formats/layer-violation-table"},
}

func OutputFormatsForFocuses(focuses []Focus, loader *promptloader.Loader) string {
	seen := make(map[string]bool)
	var paths []string

	for _, f := range focuses {
		for _, path := range focusOutputFormats[f] {
			if !seen[path] {
				seen[path] = true
				paths = append(paths, path)
			}
		}
	}

	// score-summary は常に含める
	paths = append(paths, "output-formats/score-summary")

	contents := make([]string, 0, len(paths))
	for _, path := range paths {
		if c := loader.GetContent(path); c != "" {
			contents = append(contents, c)
		}
	}
	return strings.Join(contents, "\n\n---\n\n")
}

func OutputFormatsForRefactoring(loader *promptloader.Loader) string {
	return strings.Join([]string{
		loader.GetContent("output-formats/refactoring-suggestion-table"),
		loader.GetContent("output-formats/mermaid-class-diagram"),
	}, "\n\n---\n\n")
}
